.github/workflows/deploy-docker.yaml

```diff
@@ -16,27 +20,38 @@ jobs:
 
     steps:
       - name: Check out code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Import environment variables from a file
         shell: bash
-        run: |
-          ENV_FILE="${GITHUB_WORKSPACE}/mainnet/.env"
-          while read -r line; do
-            if [[ "$line" != OBSERVER_IMAGE* ]] && [[ "$line" != PROXY_IMAGE* ]] ; then
-              continue
-            fi
-            echo "$line" >> $GITHUB_ENV
-          done < ${ENV_FILE}
-      - name: Build node image
-        run: |
-          cd ${GITHUB_WORKSPACE}/mainnet/ && docker image build . -t "${REGISTRY_HOSTNAME}/${OBSERVER_IMAGE}" -f ./chain-observer
-      - name: Build proxy image
-        run: |
-          cd ${GITHUB_WORKSPACE}/mainnet/ && docker image build . -t "${REGISTRY_HOSTNAME}/${PROXY_IMAGE}" -f ./chain-proxy
-      - name: Push images
-        run: |
-          docker login --username ${{ secrets.DOCKERHUB_USERNAME }} --password ${{ secrets.DOCKERHUB_TOKEN }}
-          docker push "${REGISTRY_HOSTNAME}/${OBSERVER_IMAGE}"
-          docker push "${REGISTRY_HOSTNAME}/${PROXY_IMAGE}"
-          docker logout
+        run: cat ${GITHUB_WORKSPACE}/mainnet/.env >> $GITHUB_ENV
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to Docker Hub
+        if: github.event_name == 'release' && github.event.action == 'published'
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push node image
+        uses: docker/build-push-action@v6
+        with:
+          context: ./mainnet
+          file: ./mainnet/chain-observer
+          platforms: linux/amd64,linux/arm64
+          push: ${{ github.event_name == 'release' && github.event.action == 'published' }}
+          tags: |
+            ${{ env.REGISTRY_HOSTNAME }}/${{ env.OBSERVER_IMAGE }}
+
+      - name: Build and push proxy image
+        uses: docker/build-push-action@v6
+        with:
+          context: ./mainnet
+          file: ./mainnet/chain-proxy
+          platforms: linux/amd64,linux/arm64
+          push: ${{ github.event_name == 'release' && github.event.action == 'published' }}
+          tags: |
+            ${{ env.REGISTRY_HOSTNAME }}/${{ env.PROXY_IMAGE }}
```

Comment:
Is this provided?

Reply:
It loads these env variables from a file on line 27:

`run: cat ${GITHUB_WORKSPACE}/mainnet/.env >> $GITHUB_ENV`